core/dbt/parser/unit_tests.py

```diff
@@ -105,44 +106,66 @@ def parse_unit_test_case(self, test_case: UnitTestDefinition):
+                "package_name": package_name,
+                "original_file_path": original_input_node.original_file_path,
+                "unique_id": f"model.{package_name}.{input_name}",
+                "config": ModelConfig(materialized="ephemeral"),
+                "database": original_input_node.database,
+                "alias": original_input_node.identifier,
+                "schema": original_input_node.schema,
+                "fqn": original_input_node.fqn,
+                "checksum": FileHash.empty(),
+            }
+
+            if original_input_node.resource_type == NodeType.Model:
+                original_input_node_columns = None
+                if original_input_node.config.contract.enforced:
+                    original_input_node_columns = {
+                        column.name: column.data_type for column in original_input_node.columns
+                    }
+                input_node = ModelNode(
+                    **common_fields,
+                    name=input_name,
+                    path=original_input_node.path,
+                    raw_code=self._build_fixture_raw_code(
+                        given.get_rows(
+                            self.root_project.project_root, self.root_project.fixture_paths
+                        ),
+                        original_input_node_columns,
                     ),
-                    original_input_node_columns,
-                ),
-                resource_type=NodeType.Model,
-                package_name=package_name,
-                path=original_input_node.path,
-                original_file_path=original_input_node.original_file_path,
-                unique_id=input_unique_id,
-                name=input_name,
-                config=ModelConfig(materialized="ephemeral"),
-                database=original_input_node.database,
-                schema=original_input_node.schema,
-                alias=original_input_node.alias,
-                fqn=input_unique_id.split("."),
-                checksum=FileHash.empty(),
-            )
+                )
+            elif original_input_node.resource_type == NodeType.Source:
+                # Note: we don't look at the source columns. Are there circumstances when we should?
+                # We are reusing the database/schema/identifier from the original source,
+                # but that shouldn't matter since this acts as an ephemeral model which just
+                # wraps a CTE around the unit test node.
+                input_node = UnitTestSourceDefinition(
+                    **common_fields,
+                    name=original_input_node.name,  # must be the same name for source lookup to work
+                    path=input_name + ".sql",  # for writing out compiled_code
+                    raw_code=self._build_fixture_raw_code(
+                        given.get_rows(
+                            self.root_project.project_root, self.root_project.fixture_paths
+                        ),
```

Comment:
Comparing this to setting [raw_code above](https://github.com/dbt-labs/dbt-core/pull/9059/files#diff-4164ae2da9549b426a9e1a37f2d15509fef9bed35de3139273a506c0ba8c8f5cR141-R144), I believe raw_code could be moved to common_fields